### PR TITLE
Wrap `items` in a `set` call for `Model.batch_get`

### PR DESCRIPTION
**Pull request by jmphilli**
Workaround to avoid triggering server validation error:
```json
{
    "Error": {
        "Message": "Provided list of item keys contains duplicates",
        "Code": "ValidationException"
    }
}
```

This doesn't update the underlying `connection.Table` or `connection.Base` `batch_get_item` call.

### CODE REVIEW

File: pynamodb/models.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def batch_get(
         :param items: Should be a list of hash keys to retrieve, or a list of
             tuples if range keys are used.
         """
-        items = list(items)
+        items = list(set(items))
```
Comment (ikonst) [inline pynamodb/models.py:327]: All we're doing with `items` is just `while items:` (== `items.__len__`) and `items.pop()`, both are supported on a set:
```suggestion
        items = set(items)
```

**Comment by ikonst** [comment]
Fixed the test: https://github.com/pynamodb/PynamoDB/pull/948

**Review by garrettheel — APPROVED** [review]

**Comment by Streadz** [comment]
+1

**Comment by sridhar562345** [comment]
Facing the below issue with batch_get, when JSONAttribute is used as hashkey 
```
File "/home/user/Desktop/project_pynamo/venv/lib/python3.7/site-packages/pynamodb/models.py", line 338, in batch_get
    items = set(items)
TypeError: unhashable type: 'dict'
```

# model
```py
class UserMatchMap(Model):
    hash_key_compound_field = JSONAttribute(
        hash_key=True,
    )
   ...
```
# sample data creation
```py
hash_key_compound_field = {
            'user_id': 1,
            'status': "A",
            'game_id': 1
}

UserMatchMap(hash_key_compound_field=hash_key_compound_field).save()
```
# Retrieving the hash keys using batch get raises the error
```py
hash_key_compound_field = {
            'user_id': 1,
            'status': "A",
            'game_id': 1
}
UserMatchMap.batch_get(items=[hash_key_compound_field])
```

If we perform set on a list of dictionaries we face the above error.

**Pull request merged**